class.jetpack-autoupdate.php

```diff
@@ -121,5 +152,123 @@ function is_version_controlled() {
 
 		return $is_version_controlled;
 	}
+
+	/**
+	 * On completion of an automatic update, let's store the results.
+	 *
+	 * @param $results - Sent by WP_Automatic_Updater after it completes an autoupdate action. Results may be empty.
+	 */
+	function automatic_updates_complete( $results ) {
+		$this->autoupdate_results = $results;
+	}
+
+	/**
+	 * On shutdown, let's check to see if we've preformed an automatic update.
+	 * If so, let's compare the expected results to the actual results, and log our findings.
+	 *
+	 * Results are logged locally via Jetpack::log(), and globally via Jetpack::do_stats()
+	 */
+	function log_results() {
+
+		if ( $this->is_updating ) {
+
+			$this->jetpack = Jetpack::init();
+			$items_to_log = array( 'plugin', 'theme' );
+
+			foreach( $items_to_log as $items ) {
+				$this->log_items( $items );
+			}
+
+			$this->jetpack->do_stats( 'server_side' );
+			$this->jetpack->log( 'autoupdates', $this->log );
+		}
+	}
+
+	/**
+	 * Iterates through expected items ( plugins or themes ) and compares them to actual results.
+	 *
+	 * @param $items 'plugin' or 'theme'
+	 */
+	function log_items( $items ) {
+		$items_updated = 0;
+		$items_failed  = 0;
+		$item_results  = $this->get_successful_updates( $items );
+
+		foreach( $this->autoupdate_expected[ $items ] as $item ) {
+			if ( in_array( $item, $item_results ) ) {
+				$items_updated++;
+				$this->log[ $items ][ $item ] = true;
+			} else {
+				$items_failed++;
+				$this->log[ $items ][ $item ] = new WP_Error( "$items-fail", $this->get_error_message( $item, $type = $items ) );
+			}
+		}
+
+		if ( $items_updated ) {
+			$this->jetpack->stat( "autoupdates/$items-success", $items_updated );
```

Comment:
$items is a string, either 'plugin' or 'theme'
poorly-named variable? yep.
array? nope.